server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1706,10 +1745,20 @@ synchronized void addShardDeleteResult(
             ShardGeneration newGeneration,
             Collection<String> blobsToDelete
         ) {
+            if (compressed == null) {
+                // No output stream: skip writing, but still update generations
```

Comment:
AFAICT this only happens if someone sets `repositories.blobstore.max_shard_delete_results_size: 0` right? I don't think we need to treat that as a special case tho, it should be very rare to do this and everything will work ok even if the limit is zero, so I'd rather we didn't have these rare `null` cases lying around nor extra code to handle them like this.